src/test/java/com/hillel/items_exchange/util/PatternHandlerTest.java

```diff
@@ -0,0 +1,302 @@
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.hillel.items_exchange.util.PatternHandler.*;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class PatternHandlerTest {
+
+    @ParameterizedTest
+    @MethodSource("createCorrectEmails")
+    void testEmailRegexp_whenEmailIsCorrect(String correctEmail) {
+        assertTrue(correctEmail.matches(EMAIL));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createWrongEmails")
+    void testEmailRegexp_whenEmailIsWrong(String wrongEmail) {
+        assertFalse(wrongEmail.matches(EMAIL));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createCorrectPasswords")
+    void testPasswordRegexp_whenPasswordsCorrect(String correctPassword) {
+        assertTrue(correctPassword.matches(PASSWORD));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createWrongPasswords")
+    void testPasswordRegexp_whenPasswordsWrong(String wrongPassword) {
+        assertFalse(wrongPassword.matches(PASSWORD));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createCorrectPhoneNumbers")
+    void testPhoneNumberRegexp_whenPhoneNumberIsCorrect(String correctPhoneNumber) {
+        assertTrue(correctPhoneNumber.matches(PHONE_NUMBER));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createWrongPhoneNumbers")
+    void testPhoneNumberRegexp_whenPhoneNumberIsWrong(String wrongPhoneNumber) {
+        assertFalse(wrongPhoneNumber.matches(PHONE_NUMBER));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createCorrectUserNames")
+    void testUsernameRegexp_whenUsernameIsCorrect(String correctUserName) {
+        assertTrue(correctUserName.matches(USERNAME));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createWrongUserNames")
+    void testUsernameRegexp_whenUsernameIsWrong(String wrongName) {
+        assertFalse(wrongName.matches(USERNAME));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createCorrectWordsEmptyOrMin2Max50")
+    void testWordEmptyOrMin2Max50_whenWordIsCorrect(String correctWord) {
+        assertTrue(correctWord.matches(WORD_EMPTY_OR_MIN_2_MAX_50));
+    }
+
+    @ParameterizedTest
+    @MethodSource("createWrongWordsEmptyOrMin2Max50")
+    void testWordEmptyOrMin2Max50_whenWordIsWrong(String wrongWord) {
+        assertFalse(wrongWord.matches(WORD_EMPTY_OR_MIN_2_MAX_50));
+    }
+
+    private static List<String> createCorrectEmails() {
+        List<String> correctEmails = new ArrayList<>();
+        correctEmails.add("pushkin@ukr.net");
+        correctEmails.add("pushkin.145@gmail.com");
+        correctEmails.add("pushkin@mail.ru");
+        correctEmails.add("pushkin.pushkin@mail.ru");
+        correctEmails.add("pushkin.pushkin@mail.mail.ru");
+        correctEmails.add("1@ukr.net");
+        correctEmails.add("1@111.ru");
+
+        return correctEmails;
+    }
```

Comment:
```suggestion
 private static List<String> createCorrectEmails() {
        return List.of(
                "pushkin@ukr.net",
                "pushkin.145@gmail.com",
                "pushkin@mail.ru",
                "pushkin.pushkin@mail.ru",
                "pushkin.pushkin@mail.mail.ru",
                "1@ukr.net",
                "1@111.ru"
        );
    }
```
You can replace such pieces with something like this. Fashionable and youthful)))

Reply:
I want to remain possibility to add some list (e.g. correctEmails.addAll(someList)) without rewriting.